deploy/21_hlg.ts

```diff
@@ -140,25 +150,48 @@ const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
     hre.deployments.log('reusing "HLG" at:', hlgTokenAddress);
   }
 
+  hre.deployments.log('checking Holograph UtilityToken reference');
   if ((await holograph.getUtilityToken()) != hlgTokenAddress) {
     const setHTokenTx = await holograph.setUtilityToken(hlgTokenAddress, {
-      nonce: await hre.ethers.provider.getTransactionCount(deployer.address),
+      ...(await txParams({
+        hre,
+        from: deployer,
+        to: holograph,
+        data: holograph.populateTransaction.setUtilityToken(hlgTokenAddress),
+      })),
     });
     await setHTokenTx.wait();
   }
 
+  hre.deployments.log('checking HolographRegistry UtilityToken reference');
   if ((await holographRegistry.getUtilityToken()) != hlgTokenAddress) {
     const setHTokenTx2 = await holographRegistry.setUtilityToken(hlgTokenAddress, {
-      nonce: await hre.ethers.provider.getTransactionCount(deployer.address),
+      ...(await txParams({
+        hre,
+        from: deployer,
+        to: holographRegistry,
+        data: holographRegistry.populateTransaction.setUtilityToken(hlgTokenAddress),
+      })),
     });
     await setHTokenTx2.wait();
   }
 
+  hre.deployments.log('checking HolographOperator HLG balance');
   if (currentNetworkType == NetworkType.testnet || currentNetworkType == NetworkType.localhost) {
     const hlgContract = (await hre.ethers.getContract('HolographERC20', deployer)).attach(hlgTokenAddress);
     if ((await hlgContract.balanceOf(operatorAddress)).isZero()) {
+      hre.deployments.log('HolographOperator has no HLG');
       const transferTx = await hlgContract.transfer(operatorAddress, BigNumber.from('1000000000000000000000000'), {
-        nonce: await hre.ethers.provider.getTransactionCount(deployer.address),
+        ...(await txParams({
```

Comment:
why 2 million eth?

Reply:
That's just gasLimit being multiplied by 2. Final result is like 140k gas units, real used amount is 78k-ish gas units.
The other stuff you're seeing is the actual erc20 token transfer transaction being encoded for extracting gasLimit with.

Reply:
As for the 1mil erc20 tx, that's for our testnets so we have tokens to use for operator rewards.